src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,107 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_NUMBER = 1.5;
+    private T[] array;
+    private int size;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void growIfArrayFull() {
+        if (size >= array.length) {
+            int newCapacity = (int) (array.length * GROWTH_NUMBER);
+            array = Arrays.copyOf(array, newCapacity);
```

Comment:
[Don't use Objects, Arrays, or any other util class.](https://github.com/mate-academy/jv-array-list/blob/master/checklist.md#dont-use-objects-arrays-or-any-other-util-class)

Reply:
I guessed, that I should have used a loop😅